src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -18,36 +19,62 @@
+
+    /**
+     * Creates a DeleteCommand to delete the specified tutor/student.
+     * @param targetIndex The specified index in either the tutor/student list, of the tutor/student to be deleted
+     * @param personType The parsed personType (only a valid input if "t" or "s")
+     */
+    public DeleteCommand(Index targetIndex, PersonType personType) {
         this.targetIndex = targetIndex;
+        this.personType = personType;
     }
 
+    /**
+     * Executes the delete command, which will remove a specified tutor/student from the list.
+     * @param model {@code Model} which the command should operate on.
+     * @return feedback message of the operation result for display
+     * @throws CommandException If an error occurs during command execution.
+     */
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        // Note: getFilteredPersonList() no longer exists.
-        // TODO: Replace with getFilteredStudentList() or getFilteredTutorList() depending on which list you want.
-        List<Person> lastShownList = model.getFilteredPersonList();
+        switch (personType) {
+        case PersonType.TUTOR:
+            List<Tutor> lastShownTutorList = model.getFilteredTutorList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
-        }
+            if (targetIndex.getZeroBased() >= lastShownTutorList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_TUTOR_DISPLAYED_INDEX);
+            }
 
-        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            Tutor tutorToDelete = lastShownTutorList.get(targetIndex.getZeroBased());
+            model.deleteTutor(tutorToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_TUTOR_SUCCESS, tutorToDelete));
+        case PersonType.STUDENT:
+            List<Student> lastShownStudentList = model.getFilteredStudentList();
 
-        // Note: deletePerson() no longer exists
-        // TODO: Change to deleteTutor() or deleteStudent() depending on the case.
-        model.deletePerson(personToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+            if (targetIndex.getZeroBased() >= lastShownStudentList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+            }
+
+            Student studentToDelete = lastShownStudentList.get(targetIndex.getZeroBased());
+            model.deleteStudent(studentToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_STUDENT_SUCCESS, studentToDelete));
+        default:
+            // Any invalid input would be handled by the DeleteCommandParser and will not reach here
+            throw new CommandException(MESSAGE_USAGE);
+        }
     }
```

Comment:
Not sure if break; is needed since you are already returning a value but according to the all mighty: https://se-education.org/guides/conventions/java/intermediate.html, break statements should always be there 🔪